src/workflows-app/components/submission-common.js

```diff
@@ -0,0 +1,588 @@
+import { Ajax } from 'src/libs/ajax';
+import { resolveWdsUrl } from 'src/libs/ajax/data-table-providers/WdsDataTableProvider';
+import colors from 'src/libs/colors';
+import { getConfig } from 'src/libs/config';
+import { notify } from 'src/libs/notifications';
+import { differenceFromDatesInSeconds, differenceFromNowInSeconds } from 'src/libs/utils';
+import * as Utils from 'src/libs/utils';
+
+export const AutoRefreshInterval = 1000 * 60; // 1 minute
+export const WdsPollInterval = 1000 * 30; // 30 seconds
+
+const iconSize = 24;
+export const addCountSuffix = (label, count = undefined) => {
+  return label + (count === undefined ? '' : `: ${count}`);
+};
+
+export const statusType = {
+  ...jobStatusType,
+  canceling: {
+    id: 'canceling', // Must match variable name for collection unpacking.
+    label: () => 'Canceling',
+    icon: (style) => icon('sync', { size: iconSize, style: { color: colors.dark(), ...style } }),
+  },
+  canceled: {
+    id: 'canceled', // Must match variable name for collection unpacking.
+    label: () => 'Canceled',
+    icon: (style) => icon('warning-standard', { size: iconSize, style: { color: colors.dark(), ...style } }),
+  },
+};
+
+/**
+ * Returns the rendered status line, based on icon function, label, and style.
+ */
+export const makeStatusLine = (iconFn, label, style) =>
+  div({ style: { display: 'flex', alignItems: 'center', fontSize: 14, ...style } }, [iconFn({ marginRight: '0.5rem' }), label]);
+
+const RunSetTerminalStates = ['ERROR', 'COMPLETE', 'CANCELED'];
+export const isRunSetInTerminalState = (runSetStatus) => RunSetTerminalStates.includes(runSetStatus);
+
+const RunTerminalStates = ['COMPLETE', 'CANCELED', 'SYSTEM_ERROR', 'ABORTED', 'EXECUTOR_ERROR'];
+export const isRunInTerminalState = (runStatus) => RunTerminalStates.includes(runStatus);
+
+const Covid19Methods = ['fetch_sra_to_bam', 'assemble_refbased', 'sarscov2_nextstrain'];
+export const isCovid19Method = (methodName) => Covid19Methods.includes(methodName);
+
+export const getDuration = (state, submissionDate, lastModifiedTimestamp, stateCheckCallback) => {
+  return stateCheckCallback(state) ? differenceFromDatesInSeconds(submissionDate, lastModifiedTimestamp) : differenceFromNowInSeconds(submissionDate);
+};
+
+export const loadAllRunSets = async (signal) => {
+  try {
+    const getRunSets = await Ajax(signal).Cbas.runSets.get();
+    const durationEnhancedRunSets = _.map(
+      (r) => _.merge(r, { duration: getDuration(r.state, r.submission_timestamp, r.last_modified_timestamp, isRunSetInTerminalState) }),
+      getRunSets.run_sets
+    );
+    return _.merge(getRunSets, { run_sets: durationEnhancedRunSets });
+  } catch (error) {
+    notify('error', 'Error getting run set data', { detail: error instanceof Response ? await error.text() : error });
+  }
+};
+
+// Invokes logic to determine the appropriate app for Cromwell
+export const resolveCromwellApp = (apps) => {
```

Comment:
I think the logic in this method might be a little outdated as WDS and Cromwell apps have now been decoupled. Instead of this method, we should use the method [resolveRunningCromwellAppUrl](https://github.com/DataBiosphere/terra-ui/blob/a83ce7514f39236d6e3efbf63c76cfc5b6a138e5/src/libs/workflows-app-utils.ts#L4) to find cbas-ui proxy url